tests/kafkatest/services/verifiable_consumer.py

```diff
@@ -0,0 +1,215 @@
+        if event["success"]:
+            for offset_commit in event["offsets"]:
+                topic = offset_commit["topic"]
+                partition = offset_commit["partition"]
+                tp = TopicPartition(topic, partition)
+                self.committed_offsets[tp] = offset_commit["offset"]
+
+    def _handle_records_consumed(self, node, event):
+        for topic_partition in event["partitions"]:
+            topic = topic_partition["topic"]
+            partition = topic_partition["partition"]
+            tp = TopicPartition(topic, partition)
+            self.consumed_positions[tp] = topic_partition["maxOffset"] + 1
+        self.total_records += event["count"]
+
+    def _handle_partitions_revoked(self, node, event):
+        self.revoked_count += 1
+        self.assignment[node] = []
+        if node in self.joined:
+            self.joined.remove(node)
+
+    def _handle_partitions_assigned(self, node, event):
+        self.assigned_count += 1
+        self.joined.add(node)
+        assignment =[]
+        for topic_partition in event["partitions"]:
+            topic = topic_partition["topic"]
+            partition = topic_partition["partition"]
+            assignment.append(TopicPartition(topic, partition))
+        self.assignment[node] = assignment
+
+    def start_cmd(self, node):
+        cmd = ""
+        cmd += "export LOG_DIR=%s;" % VerifiableConsumer.LOG_DIR
+        cmd += " export KAFKA_OPTS=%s;" % self.security_config.kafka_opts
+        cmd += " export KAFKA_LOG4J_OPTS=\"-Dlog4j.configuration=file:%s\"; " % VerifiableConsumer.LOG4J_CONFIG
+        cmd += "/opt/" + kafka_dir(node) + "/bin/kafka-run-class.sh org.apache.kafka.tools.VerifiableConsumer" \
+              " --groupId %s --topic %s --broker-list %s --session-timeout %s" % \
+              (self.group_id, self.topic, self.kafka.bootstrap_servers(), self.session_timeout)
+        if self.max_messages > 0:
+            cmd += " --max-messages %s" % str(self.max_messages)
+
+        cmd += " --consumer.config %s" % VerifiableConsumer.CONFIG_FILE
+        cmd += " 2>> %s | tee -a %s &" % (VerifiableConsumer.STDOUT_CAPTURE, VerifiableConsumer.STDOUT_CAPTURE)
+        print(cmd)
+        return cmd
+
+    def pids(self, node):
+        try:
+            cmd = "jps | grep -i VerifiableConsumer | awk '{print $1}'"
+            pid_arr = [pid for pid in node.account.ssh_capture(cmd, allow_fail=True, callback=int)]
+            return pid_arr
+        except (subprocess.CalledProcessError, ValueError) as e:
+            return []
+
+    def try_parse_json(self, string):
+        """Try to parse a string as json. Return None if not parseable."""
+        try:
+            record = json.loads(string)
+            return record
+        except ValueError:
+            self.logger.debug("Could not parse as json: %s" % str(string))
+            return None
+
+    def kill_consumer(self, node, clean_shutdown=True, allow_fail=False):
```

Comment:
Why do you need separate `kill_consumer` method and a `stop_node` method?
Or maybe just make the naming consistent with your change to `verifiable_producer.py` and call this `kill_node`